src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiRegularIndex.java

```diff
@@ -112,59 +83,183 @@ public ApiRegularIndex create(
 
       var apiColumnDef = checkIndexColumnExists(tableSchemaObject, targetIdentifier);
 
+      // create an ApiRegularIndex for the target primitive datatype
+      if (apiColumnDef.type().isPrimitive()) {
+        return createApiIndexForPrimitive(
+            tableSchemaObject, apiColumnDef, indexIdentifier, targetIdentifier, indexDesc);
+      }
+
+      // create an ApiRegularIndex for the target map/set/list datatype
+      if (apiColumnDef.type().isContainer()
+          && apiColumnDef.type().typeName() != ApiTypeName.VECTOR) {
+        try {
+          return createApiIndexForCollection(
+              tableSchemaObject, apiColumnDef, indexIdentifier, targetIdentifier, indexDesc);
+        } catch (UnknownCqlIndexFunctionException e) {
+          // This won't happen, since the index function has already been validated.
+        }
+      }
+
       // we could check if there is an existing index but that is a race condition, we will need to
       // catch it if it fails
-      // regular indexes can only be on primitive. Adding indexes on maps, sets, lists will come
-      // later.
-      if (!apiColumnDef.type().isPrimitive()) {
-        throw SchemaException.Code.UNSUPPORTED_INDEXING_FOR_DATA_TYPES.get(
+      throw SchemaException.Code.UNSUPPORTED_INDEXING_FOR_DATA_TYPES.get(
```

Comment:
the error message for UNSUPPORTED_INDEXING_FOR_DATA_TYPES says the data type is frozen, but there are no checks I can see for that.